tests/unit/test_data.py

```diff
@@ -9,3 +14,42 @@ def test_repeating_loader():
         assert next(loader) == 1
         assert next(loader) == 2
         assert next(loader) == 3
+
+
+def test_dataloader_drop_last(tmpdir):
+    config_dict = {
+        "train_batch_size": 1,
+        "steps_per_print": 1, 
+    }
+    args = args_from_dict(tmpdir, config_dict)
+    hidden_dim = 10
+
+    model = SimpleModel(hidden_dim)
+
+    def random_dataloader(model, total_samples, hidden_dim, device, drop_last, dtype=torch.half):
+        batch_size = model.train_micro_batch_size_per_gpu()
+        train_data = torch.randn(total_samples, hidden_dim, device=device, dtype=dtype)
+        train_label = torch.empty(total_samples,
+                                dtype=torch.long,
+                                device=device).random_(hidden_dim)
+        train_dataset = torch.utils.data.TensorDataset(train_data, train_label)
```

Comment:
Can you please refactor this into a `random_dataset()` in `test/unit/simple_model.py`?

Reply:
@tjruwase I am sorry but I couldn't understand your comment.

The [random_dataloader()](https://github.com/microsoft/DeepSpeed/blob/18d89c62f26c428953090aebc84c858b77fcaebd/tests/unit/test_data.py#L29-L37) in tests/unit/test_data.py is actually refactored from [random_dataloader()](https://github.com/microsoft/DeepSpeed/blob/274c375c87f35d1c7509dfc6a14f8a84d5efab84/tests/unit/simple_model.py#L181-L189) in test/unit/simple_model.py. 

The changes I made there is to add the `dataloader_drop_last = True` Argument

or is it that you need me to use 
```
train_loader = torch.utils.data.DataLoader(train_dataset, batch_size=batch_size) 
```
instead of 
```
train_loader = DeepSpeedDataLoader(train_dataset,
                                           batch_size=batch_size,
                                           dataloader_drop_last=drop_last)
```

Please correct me if I am wrong.

Reply:
Sorry for not being clear. What I meant is not to reimplement [random_dataloader()](https://github.com/microsoft/DeepSpeed/blob/6eb13a5245599cc5f76fec72aee0d88edfaeeb04/tests/unit/test_data.py#L29) in test_data.py, which essentially duplicates [random_dataloader()](https://github.com/microsoft/DeepSpeed/blob/336dd089e5e11d46daf3abfad048d71b72912504/tests/unit/simple_model.py#L181) in simple_model.py. Instead, why not do the following

1. In simple_model.py, implement random_dataset() to provide [these](https://github.com/microsoft/DeepSpeed/blob/336dd089e5e11d46daf3abfad048d71b72912504/tests/unit/simple_model.py#L182-L187) core logic of random_dataloader().
2. In simple_model.py, use the random_dataset() in random_dataloader() to avoid duplication.
3. In test_data.py, delete random_dataloader(), and use random_dataset() directly in _test_dataloader_drop_last() such as below:
```python
train_dataset = random_dataset(model, ...)
data_loader = DeepSpeedDataLoader(train_dataset, ...)
```
Will that work? Thanks.

Reply:
@tjruwase 

Thanks a lot for the detailed explanation. I have made the changes. Please review.